src/robotFactory.js

```diff
@@ -1,13 +1,95 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: \
+${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  }
```

Comment:
You can use rest and spread operators here.

```suggestion
  constructor(...args) {
    super(...args);
    this.coords.z = coords.z || 0;
  }
```

Reply:
I dont understand how will we get coords? coords.z will be is not defined if we will use ...arg.

Reply:
class FlyingRobot extends BaseRobot {
  constructor(...args) {
    super(...args);

  const [name, weight, coords, chipVersion] = args;

    this.coords.z = coords.z || 0;
  }
 i can make something like this, but i think it`l be worse version than now and linter gives errors because we dont use name, weight, chipVersion.